bugsnag.go

```diff
@@ -25,94 +28,107 @@ type BugsnagReporter struct {
 
 	Backup ErrorReporter
 }
+type bugsnagMetadata struct {
+	errorClass    string
+	context       string
+	groupingHash  string
+	severity      string
+	eventMetadata *hashstruct.Hash
+}
 
-func (er *BugsnagReporter) Report(ctx context.Context, newErr error) {
-	depth := 2
-	if IsNestedReporter(ctx) {
-		depth = 1
-	}
-
-	payload := er.newPayload(depth, newErr)
+func (er *BugsnagReporter) Report(ctx context.Context, newErr *error.Error, metadata *bugsnagMetadata) {
```

Comment:
This is backwards incompatible change to the API right? Also I think not always you'd need meta data, so I'm wondering whether make another exportable function like ReportWithMetaData would be better choice, wdyt?